### PR TITLE
core/types: support yParity field in JSON transactions

### LINKED ISSUE #27727: Type 1 and type 2 transactions have `yParity`, not `v`
https://github.com/ethereum/go-ethereum/blob/34d507215951fb3f4a5983b65e127577989a6db8/internal/ethapi/api.go#L1375

Geth currently is returning transactions that have a `v` value even when that doesn't make sense.  EIP-2930 and EIP-1559 declare a `yParity` property on those transactions which is defined as "The parity (0 for even, 1 for odd) of the y-value of a secp256k1 signature."  `v` is not defined anywhere in either of those EIPs and what one might put in a variable called `v` is incredibly ambiguous as it could be any of:
1. yParity + 27
2. yParity + 35 + 2 * chainId
3. yParity (what I think geth is doing)

While I advocate for fixing the source code to use `yParity` instead of `v` for type 1 and 2 transactions internally so the code is more readable and follows the specification closer, what actually matters here is that the JSON-RPC should be including a `yParity` in any transaction response with a type 1 or type 2 transaction.

For backward compatibility, it seems reasonable to continue to include a `v` value that contains whatever it contains now, with perhaps some plan (or not) to eventually remove it.

### PR DESCRIPTION
**Pull request by fjl**
Fixes #27727

### CODE REVIEW

**Review by lightclient — COMMENTED** [review]
Looks good in general, just a couple small questions.

File: core/types/transaction_marshalling.go [hunk 1]
```diff
@@ -155,23 +186,24 @@ func (tx *Transaction) UnmarshalJSON(input []byte) error {
 			return errors.New("missing required field 'input' in transaction")
 		}
 		itx.Data = *dec.Input
-		if dec.V == nil {
```
[the remark this answers is not in the document]
Comment (fjl) [reply core/types/transaction_marshalling.go:158]: Seemed to look better to me.

File: core/types/transaction_marshalling.go [hunk 2]
```diff
@@ -155,23 +186,24 @@ func (tx *Transaction) UnmarshalJSON(input []byte) error {
...
 			return errors.New("missing required field 'r' in transaction")
 		}
 		itx.R = (*big.Int)(dec.R)
+		// signature S
 		if dec.S == nil {
 			return errors.New("missing required field 's' in transaction")
 		}
 		itx.S = (*big.Int)(dec.S)
-		withSignature := itx.V.Sign() != 0 || itx.R.Sign() != 0 || itx.S.Sign() != 0
-		if withSignature {
```
[the remark this answers is not in the document]
Comment (fjl) [reply core/types/transaction_marshalling.go:171]: I have restored this behavior now, but I don't really get it. The signature values are mandatory, but all-zero is OK?

[the remark this answers is not in the document]
Comment (lightclient) [reply core/types/transaction_marshalling.go:171]: The only place I am aware this functionality is used is in t8n to denote that the transaction should be signed using the provided `secretKey`: https://github.com/ethereum/go-ethereum/blob/7f756dc1185d7f1eeeacb1d12341606b7135f9ea/cmd/evm/testdata/8/txs.json#L19-L21

Would probably be good to get rid of all-zero values at some point, but I'm not sure what else depends on this behavior.

File: core/types/transaction_marshalling.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (lightclient) [reply core/types/transaction_marshalling.go:158]: Feels weird to have them out of order. `v`, `r`, `s` ordering is pretty widely understood.

File: core/types/transaction_marshalling.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (fjl) [reply core/types/transaction_marshalling.go:171]: Either way it doesn't matter since an all zero signature will be invalid anyway.

**Review by holiman — APPROVED** [review]
LGTM

**Pull request merged**